halo2-base/src/lib.rs

```diff
@@ -197,19 +248,23 @@ impl<F: ScalarField> Context<F> {
         AssignedValue { value: self.advice[offset], cell }
     }
 
+    /// Creates an equality constraint between two `advice` cells.
+    /// * `a`: the first `advice` cell to be constrained equal
+    /// * `b`: the second `advice` cell to be constrained equal
+    /// * Assumes both cells are `advice` cells
     pub fn constrain_equal(&mut self, a: &AssignedValue<F>, b: &AssignedValue<F>) {
         if !self.witness_gen_only {
             self.advice_equality_constraints.push((a.cell.unwrap(), b.cell.unwrap()));
         }
     }
 
-    /// Assigns multiple advice cells and the accompanying selector cells.
-    ///
-    /// Returns the slice of assigned cells.
+    /// Pushes multiple advice cells to the `advice` column of [Context] and enables them by enabling the corresponding selector specified in `gate_offset`.
+    /// 
+    /// Assumes `inputs` and `gate_offsets` are the same length.
```

Comment:
`inputs` and `gate_offset` do *not* need to be the same length. For example `inputs` can be length 4 because we could have `gate_offset = [0]` to specify enabling a single gate for these 4 cells.